### PR TITLE
Support bracketed IPv6 addresses in Host header

### PR DESCRIPTION
**Pull request by pieterlexis**
Port of https://github.com/PowerDNS/pdns/pull/8815

### CODE REVIEW

**Comment by cmouse** [comment]
Would it be possible to have unit test for this?

**Comment by pieterlexis** [comment]
image

but seriously, I wrote this code a long time ago, so I have to get into it again to figure that out

**Comment by cmouse** [comment]
It should be fairly simple as there is already unit tests for requests in tests/ folder.

File: tests/test-request.cpp [hunk 1]
```diff
@@ -200,3 +200,15 @@ User-Agent: YaHTTP v1.0\r\n\r\n");
...
+BOOST_AUTO_TEST_CASE(test_get_host_ipv6_literal_with_port) 
+{
+YaHTTP::Request req;
+req.version = 11;
+req.setup("GET", "http://[2001:db8::1]:5555/test");
+req.GET()["hello"] = "world";
+req.GET()["tree"] = "apple";
+BOOST_CHECK_EQUAL(req.str(),
+"GET /test?hello=world&tree=apple HTTP/1.1\r\n\
+Host: [2001:db8::1]\r\n\
```
Comment (cmouse) [inline tests/test-request.cpp:212]: shouldn't there be a port here, since it's non-standard port?

Comment (pieterlexis) [reply tests/test-request.cpp:212]: hmm yes, although I remember I didn't look at the request class actually _doing_ a request.. just ensuring a request is parsed correctly. So

1. This test is not testing my change
2. This is a different bug in yahttp

**Pull request merged**